rosdiagnostic/src/rosdiagnostic/rosdiagnostic.py

```diff
@@ -0,0 +1,200 @@
+        if not self._options.follow:
+            rospy.signal_shutdown('Run complete')
+
+    def _diag_callback(self, msg):
+        prog = re.compile(self._options.filter)
+        self._print_begin_banner(msg)
+        for diag in sorted(self._get_leaf_statuses(msg.status), key=lambda d: d.level, reverse=True):
+            if prog.search(diag.name):
+                self._print_status(diag)
+        self._print_end_banner(msg)
+
+    def _print_status(self, diag):
+        if self._options.min_level <= diag.level:
+            print('[ {} ] {} - {}'.format(self._lvl_strings_map[diag.level], diag.name, diag.message))
+            if self._options.detail:
+                print('    hardware_id: {}'.format(diag.hardware_id))
+                for kv in diag.values:
+                    print('    - {}: {}'.format(kv.key, kv.value))
+
+
+def _rosdiagnostic_cmd_echo(argv):
+    parser = argparse.ArgumentParser(description='ROS Diagnostic Viewer')
+    parser.add_argument('-f', '--follow', action='store_true', dest='follow',
+                        help='follows the diagnostic messages continuously')
+
+    parser.add_argument('-l', action='store', metavar='LEVEL', type=int, default=DiagnosticStatus.WARN,
+                        dest='min_level',
+                        help='the minimum diagnostic level to display (OK=0, WARN=1, ERROR=2, STALE=3) [default=1]')
+
+    parser.add_argument('--namespace', action='store', metavar='NAMESPACE', type=str, default='~', dest='ns',
+                        help='root namespace where to look for the "/diagnostics_agg" topic')
+
+    parser.add_argument('--filter', action='store', metavar='FILTER', type=str, default='.*', dest='filter',
+                        help='regular expression to be applied as a filter to the diagnostic name')
+
+    parser.add_argument('--nocolor', action='store_true',
+                        help='output should not make use of any color')
+
+    parser.add_argument('--detail', action='store_true',
+                        help='printing the full diagnostic details ')
+
+    args = parser.parse_args(argv)
+
+    _check_master()
+    rospy.init_node('rosdiagnostic', anonymous=True)
+    rosdiag = RosDiagnostics(args)
+    rospy.spin()
+    del rosdiag
+
+
+def _fullusage():
+    print("""rosdiagnostic is a command-line tool for printing information about ROS Diagnostics.
+
+Commands:
+\trosdiagnostic echo\tprint diagnostic messages to screen
+
+Type rosdiagnostic <command> -h for more detailed usage, e.g. 'rosdiagnostic echo -h'
+""")
+    sys.exit(getattr(os, 'EX_USAGE', 1))
+
+
+def rosdiagnosticmain(argv=None):
+    if argv is None:
+        argv = sys.argv
+    # filter out remapping arguments in case we are being invoked via roslaunch
+    argv = rospy.myargv(argv)
+
+    # process argv
+    if len(argv) == 1:
+        _fullusage()
+    try:
+        command = argv[1]
+        if command == 'echo':
```

Comment:
Now that there's only one command, this seems like unnecessary typing.

I think this would be more compact and easier to use if the functionality of `_fullusage` was integrated into the ArgumentParser instance and all of the logic for switching commands was removed.